lrpar/src/lib/ctbuilder.rs

```diff
@@ -618,3 +618,67 @@ pub fn _reconstitute<'a, StorageT: Deserialize<'a> + Hash + PrimInt + Unsigned>(
     let stable = deserialize(stable_buf).unwrap();
     (grm, sgraph, stable)
 }
+
+#[cfg(test)]
+mod test {
+
+    extern crate temp_testdir;
+    use std::path::PathBuf;
+    use std::fs::File;
+    use std::io::Write;
+
+    use self::temp_testdir::TempDir;
+    use super::{CTParserBuilder, ActionKind, CTConflictsError};
+
+    #[test]
+    fn test_conflicts() {
+
+        let temp = TempDir::default();
+        let mut file_path = PathBuf::from(temp.as_ref());
+        file_path.push("grm.y");
+        let mut f = File::create(file_path.clone()).unwrap();
+        let _ = f.write_all("%start A
+%%
+A : 'a' 'b' | B 'b';
+B : 'a' | C;
+C : 'a';".as_bytes());
+
+        match CTParserBuilder::<u8>::new()
+            .error_on_conflicts(false)
+            .action_kind(ActionKind::GenericParseTree)
+            .process_file_in_src(file_path.to_str().unwrap()) {
+            Ok((_, conflicts)) => {
+                assert_eq!(conflicts.as_ref().unwrap().stable.conflicts().unwrap().sr_len(), 1);
```

Comment:
Addressing https://github.com/softdevteam/grmtools/pull/55#pullrequestreview-181277687 (assuming it's possible) will remove at least one `as_ref`.

Reply:
I'm not sure what you mean. Your link is related to statetable, here we don't have statetable but `process_in_src` returns an `Option<Conflicts>` which is moved (we can't return a reference, since the value will be out of scope once we return from the function).

Reply:
Mea culpa!

However, this API is borderline unusable: `as_ref().unwrap().stable.conflicts().unwrap().sr_len()` is almost impossible to remember. Can we do *something* to make things easier? One thing might be to not expose `stable`, but instead expose things like `sr_len`? But that might cause us the same duplication-of-methods problem we had earlier in the PR...

Reply:
I can give the `CTConflictsError` struct some functions, at the very least `fn conflicts() -> Conflicts`.

Reply:
That would certainly help!